src/main/java/org/opensearch/flowframework/workflow/UpdateIndexStep.java

```diff
@@ -0,0 +1,190 @@
+                Map<String, Object> sourceAsMap = XContentHelper.convertToMap(configurationsBytes, false, MediaTypeRegistry.JSON).v2();
+
+                // TODO : Add support to update index mappings
+
+                // extract index settings from configuration
+                if (!sourceAsMap.containsKey("settings")) {
+                    String errorMessage = "Failed to update index settings for index "
+                        + indexName
+                        + ", settings are not found in the index configuration";
+                    throw new FlowFrameworkException(errorMessage, RestStatus.BAD_REQUEST);
+                } else {
+
+                    @SuppressWarnings("unchecked")
+                    Map<String, Object> updatedSettings = (Map<String, Object>) sourceAsMap.get("settings");
+
+                    // check if settings are flattened or expanded
+                    Map<String, Object> flattenedSettings = new HashMap<>();
+                    if (updatedSettings.containsKey("index")) {
+                        flattenSettings("", updatedSettings, flattenedSettings);
+                    } else {
+                        flattenedSettings.putAll(updatedSettings);
+                    }
+
+                    Map<String, Object> filteredSettings = new HashMap<>();
+
+                    // Retrieve current Index Settings
+                    GetSettingsRequest getSettingsRequest = new GetSettingsRequest();
+                    getSettingsRequest.indices(indexName);
+                    getSettingsRequest.includeDefaults(true);
+                    client.admin().indices().getSettings(getSettingsRequest, ActionListener.wrap(response -> {
+                        Map<String, Settings> indexToSettings = new HashMap<String, Settings>(response.getIndexToSettings());
+
+                        // Include in the update request only settings with updated values
+                        Settings currentIndexSettings = indexToSettings.get(indexName);
+                        for (Map.Entry<String, Object> e : flattenedSettings.entrySet()) {
+                            String val = e.getValue().toString();
+                            if (!val.equals(currentIndexSettings.get(e.getKey()))) {
+                                filteredSettings.put(e.getKey(), e.getValue());
+                            }
+                        }
+                    }, ex -> {
+                        Exception e = getSafeException(ex);
+                        String errorMessage = (e == null ? "Failed to update the index settings for index " + indexName : e.getMessage());
+                        updateIndexFuture.onFailure(new WorkflowStepException(errorMessage, ExceptionsHelper.status(e)));
+                    }));
+
+                    updateSettingsRequest.settings(filteredSettings);
+                }
+            }
+
+            client.admin().indices().updateSettings(updateSettingsRequest, ActionListener.wrap(acknowledgedResponse -> {
```

Comment:
Don't we want to check if filteredSettings is not empty before making the update call?

Reply:
good catch, adding this check in the next commit